src/odin/directionsbuilder.cc

```diff
@@ -101,23 +103,41 @@ TripDirections DirectionsBuilder::Build(const DirectionsOptions& directions_opti
 
 // Update the heading of ~0 length edges.
 void DirectionsBuilder::UpdateHeading(EnhancedTripPath* etp) {
+  auto is_walkway = [](TripPath_Use use) -> bool {
+    return ((use == TripPath_Use_kSidewalkUse) || (use == TripPath_Use_kFootwayUse) ||
+            (use == TripPath_Use_kStepsUse) || (use == TripPath_Use_kPathUse) ||
+            (use == TripPath_Use_kPedestrianUse) || (use == TripPath_Use_kBridlewayUse));
```

Comment:
Also, what about `Use_kOtherUse` and `TripPath_Use_kFerryUse`? Do we need them?

Reply:
These are not specific for walking - we do not need them here